code/ship/ship.cpp

```diff
@@ -4125,34 +4885,34 @@ int get_default_player_ship_index()
 // Goober5000 - this works better in its own function
 void ship_set_default_player_ship()
 {
-	int i;
-
 	// already have one
 	if(strlen(default_player_ship))
 		return;
 
+	auto it = Ship_info.cbegin();
```

Comment:
Every compiler should be able to optimize the code. The "declare once and reuse" rule may be accurate for C but it certainly isn't for C++.